tests/artifacts/test_artifacts.py

```diff
@@ -357,6 +357,45 @@ def test_log_artifact_with_target_path_and_upload_options(target_path, upload_op
         assert logged_artifact.metadata.hash is None
 
 
+@pytest.mark.parametrize(
+    "artifact_key,expected",
+    [
+        ("artifact@key", pytest.raises(mlrun.errors.MLRunInvalidArgumentError)),
+        ("artifact#key", pytest.raises(mlrun.errors.MLRunInvalidArgumentError)),
+        ("artifact!key", pytest.raises(mlrun.errors.MLRunInvalidArgumentError)),
+        ("artifact_key!", pytest.raises(mlrun.errors.MLRunInvalidArgumentError)),
+        ("artifact/key", pytest.raises(mlrun.errors.MLRunInvalidArgumentError)),
+        ("artifact_key123", does_not_raise()),
+        ("artifact-key", does_not_raise()),
+        ("artifact.key", does_not_raise()),
```

Comment:
Why include all these tests if they are already validated in t`est_validate_artifact_name`? 
I think you could use just one invalid artifact key here and modify `test_validate_artifact_name` to also test `db_key`, not just `artifact.key`

Reply:
this test is intended to test the flow of log_artifact, so i think it is okay to test it with some invalid keys.
and also it tests log_artifact when db_key != key